packages/embark/src/lib/modules/contracts_manager/index.js

```diff
@@ -466,18 +487,21 @@ class ContractsManager {
       function removeContractsWithNoCode(callback) {
         let className, contract;
         let dictionary = Object.keys(self.contracts);
-        for (className in self.contracts) {
+
+        dictionary.forEach(className => {
           contract = self.contracts[className];
 
           if (contract.code === undefined && !contract.abiDefinition) {
-            self.logger.error(__("%s has no code associated", className));
-            let suggestion = utils.proposeAlternative(className, dictionary, [className]);
-            if (suggestion) {
-              self.logger.warn(__('did you mean "%s"?', suggestion));
+            if (self.compiledContracts[className]) {
+              self.logger.error(__("%s has no code associated", className));
```

Comment:
We wanna only output this message if the Smart Contract in question had actually been compiled.

Reply:
Add a comment in the source please :)

Reply:
Will do! 👌

Reply:
Done